src/Symfony/Component/Notifier/Transport/Dsn.php

```diff
@@ -104,13 +93,18 @@ public function getRequiredOption(string $key)
         return $this->options[$key];
     }
 
+    public function getOptions(): array
+    {
+        return $this->options;
+    }
+
     public function getPath(): ?string
     {
         return $this->path;
     }
 
     public function getOriginalDsn(): string
```

Comment:
Nitpick: What about renaming it to `__toString()`?

Reply:
I would avoid that, because that would be yet another BC break